source/lib/ze_lib.cpp

```diff
@@ -401,6 +404,81 @@ zelSetDelayLoaderContextTeardown()
     #endif
 }
 
+#ifdef DYNAMIC_LOAD_LOADER
+void stabilityCheck(std::promise<int> stabilityPromise) {
```

Comment:
Please add some basic docs around the expectations of the thread and when it would be killed or dies.

Reply:
updated in the latest